### ISSUE TITLE
about the test results

### ISSUE THREAD

**Issue by sx1999** [yellow-binary-tree/HawkEye#5]
thanks for your great work!
i use the checkpoint in [https://huggingface.co/wangyueqian/HawkEye/tree/main](url) to test grounding performance, but the results i get have a large margin compare with the paper. May I ask what could be the reason for such poor results?
![image](https://github.com/yellow-binary-tree/HawkEye/assets/58420256/486a8e15-86f8-469e-b51a-53762838cbfb)
![image](https://github.com/yellow-binary-tree/HawkEye/assets/58420256/902f278e-9aef-4165-8609-72434b24b2e4)

in addition，could you share with me the activitynet dataset you used to test the ActivityNet-Captions grounding results, for the activitynet dataset i get is missing some part compare with the test-anno/anetc-recursive_grounding.json

**Comment by yellow-binary-tree** [comment]
Can you provide some details of your reproduction?

For example, the output log when executing `tasks/test_recursive_grounding.py`, and better provide the md5sum value of your checkpoints (for example, `videochat2_7b_stage2.pth`, `umt_l16_qformer.pth`, `hawkeye.pth` and the vicuna-7b-v0 checkpoints), so I can further help you with your issue.

**Comment by sx1999** [comment]
> Can you provide some details of your reproduction?
> 
> For example, the output log when executing `tasks/test_recursive_grounding.py`, and better provide the md5sum value of your checkpoints (for example, `videochat2_7b_stage2.pth`, `umt_l16_qformer.pth`, `hawkeye.pth` and the vicuna-7b-v0 checkpoints), so I can further help you with your issue.

thanks for your reply

the output log is here
[charades_sta-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077015/charades_sta-recursive_grounding-4_turns.log)
[anetc-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077023/anetc-recursive_grounding-4_turns.log)
[nextgqa-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077025/nextgqa-recursive_grounding-4_turns.log)
and the md5sum value of my checkpoints are:
d39659f971a25b08c83c408d7c1a5de8  videochat2_7b_stage2.pth
92bbcbcad4698df94bc2ca06a343f456  umt_l16_qformer.pth
bec59ffc86e15a5ddfb2a77af9d989bb  hawkeye.pth
the vicuna-7b-v0 checkpoint is converted by python src/transformers/models/llama/convert_llama_weights_to_hf.py and python3 apply_delta.py and the md5sum values are:
4ae1a6135143c0a54096438c722e6a4b  vicuna-7b-v0/model-00001-of-00003.safetensors
5466a90989d3ba56468b784c336ac90b  vicuna-7b-v0/model-00002-of-00003.safetensors
423ebf852a65b2eacc6aab86af94b858  vicuna-7b-v0/model-00003-of-00003.safetensors

looking forward to your reply

**Comment by yellow-binary-tree** [comment]
> the output log is here [charades_sta-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077015/charades_sta-recursive_grounding-4_turns.log) [anetc-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077023/anetc-recursive_grounding-4_turns.log) [nextgqa-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077025/nextgqa-recursive_grounding-4_turns.log) and the md5sum value of my checkpoints are: d39659f971a25b08c83c408d7c1a5de8 videochat2_7b_stage2.pth 92bbcbcad4698df94bc2ca06a343f456 umt_l16_qformer.pth bec59ffc86e15a5ddfb2a77af9d989bb hawkeye.pth the vicuna-7b-v0 checkpoint is converted by python src/transformers/models/llama/convert_llama_weights_to_hf.py and python3 apply_delta.py and the md5sum values are: 4ae1a6135143c0a54096438c722e6a4b vicuna-7b-v0/model-00001-of-00003.safetensors 5466a90989d3ba56468b784c336ac90b vicuna-7b-v0/model-00002-of-00003.safetensors 423ebf852a65b2eacc6aab86af94b858 vicuna-7b-v0/model-00003-of-00003.safetensors
> 
> looking forward to your reply

I think I've found the reason. You should use the `config.json` from huggingface repo [https://huggingface.co/wangyueqian/HawkEye/blob/main/config.json](https://huggingface.co/wangyueqian/HawkEye/blob/main/config.json), instead of `configs/config.json` in this github repo.
The main difference is, in `configs/config.json` `model.use_lora` is set to false, so the lora weights of the LLM wasn't loaded, which is shown in "unexpected_keys" in line 109 in the log files. While in `config.json` from huggingface, `model.use_lora` is set to true, which is the correct setting.

I apologize for my negligence here and will fix this problem ASAP.

**Comment by sx1999** [comment]
> > the output log is here [charades_sta-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077015/charades_sta-recursive_grounding-4_turns.log) [anetc-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077023/anetc-recursive_grounding-4_turns.log) [nextgqa-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15077025/nextgqa-recursive_grounding-4_turns.log) and the md5sum value of my checkpoints are: d39659f971a25b08c83c408d7c1a5de8 videochat2_7b_stage2.pth 92bbcbcad4698df94bc2ca06a343f456 umt_l16_qformer.pth bec59ffc86e15a5ddfb2a77af9d989bb hawkeye.pth the vicuna-7b-v0 checkpoint is converted by python src/transformers/models/llama/convert_llama_weights_to_hf.py and python3 apply_delta.py and the md5sum values are: 4ae1a6135143c0a54096438c722e6a4b vicuna-7b-v0/model-00001-of-00003.safetensors 5466a90989d3ba56468b784c336ac90b vicuna-7b-v0/model-00002-of-00003.safetensors 423ebf852a65b2eacc6aab86af94b858 vicuna-7b-v0/model-00003-of-00003.safetensors
> > looking forward to your reply
> 
> I think I've found the reason. You should use the `config.json` from huggingface repo https://huggingface.co/wangyueqian/HawkEye/blob/main/config.json, instead of `configs/config.json` in this github repo. The main difference is, in `configs/config.json` `model.use_lora` is set to false, so the lora weights of the LLM wasn't loaded, which is shown in "unexpected_keys" in line 109 in the log files. While in `config.json` from huggingface, `model.use_lora` is set to true, which is the correct setting.
> 
> I apologize for my negligence here and will fix this problem ASAP.

thanks for your reply!
I changed the config.json, and the results seems a little better, but still not comparable with the paper, and the mvbench result is a little lower than the paper's.
here are the results and logs
![image](https://github.com/yellow-binary-tree/HawkEye/assets/58420256/e1a69d91-52a5-41d1-b5d2-44a8822639dd)
![image](https://github.com/yellow-binary-tree/HawkEye/assets/58420256/c568ac63-e22e-4569-8123-bcc3ae73442b)
[charades_sta-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15085376/charades_sta-recursive_grounding-4_turns.log)
[nextgqa-recursive_grounding-4_turns.log](https://github.com/yellow-binary-tree/HawkEye/files/15085377/nextgqa-recursive_grounding-4_turns.log)
and here are mvbench results and log
{'Moving Direction': 25.5, 'Action Localization': 35.5, 'Scene Transition': 78.5, 'Action Count': 33.5, 'Moving Count': 43.5, 'Moving Attribute': 55.50000000000001, 'State Change': 44.0, 'Fine-grained Pose': 31.5, 'Action Sequence': 46.0, 'Action Prediction': 45.0, 'Action Antonym': 68.5, 'Fine-grained Action': 40.0, 'Unexpected Action': 51.5, 'Object Existence': 57.49999999999999, 'Object Interaction': 59.5, 'Object Shuffle': 30.0, 'Character Order': 41.0, 'Egocentric Navigation': 31.5, 'Episodic Reasoning': 29.5, 'Counterfactual Inference': 50.0, 'Avg': 44.875}
[MVBench-hawkeye.log](https://github.com/yellow-binary-tree/HawkEye/files/15085409/MVBench-hawkeye.log)

could you help me check it?
thanks a lot!

**Comment by yellow-binary-tree** [comment]
Thank you for your issue, and I found a bug in `models/hawkeye_it.py`, when cleaning the code for open source, I accidentally removed the the code for loading `videochat2_7b_stage2.pth`, which resulted in the ViT model not loaded correctly😭

I will carefully check and test the code to ensure that the results can be completely reproduced, but this may take a few days. If you're eager to reproduce our results, you can modify these lines by replacing `stage3_model_path` with `config.videochat2_model_path` (the path to `videochat2_7b_stage2.pth`) and try again:

https://github.com/yellow-binary-tree/HawkEye/blob/358cec424c3224412034e0a238dbeed4800a1c7d/models/hawkeye_it.py#L185-L187

We deeply apologize for any inconvenience this may have caused😭

**Comment by sx1999** [comment]
thank you very much! maybe I get the right results with your help. but the results seem a little different with the paper.
![image](https://github.com/yellow-binary-tree/HawkEye/assets/58420256/b4806036-d58e-498f-adef-19349961fe28)
![image](https://github.com/yellow-binary-tree/HawkEye/assets/58420256/e49fe404-d5f9-4463-8785-8cb6291f2f5a)
could you tell me the results you report is which turn of turns 1,2,3,4？it seems not be mentioned in the paper. and how should we select the finally result from the four results, latest one or best one?

**Comment by yellow-binary-tree** [comment]
the `turns: 1, 2, 3, 4` denotes the `max_rounds` hyper-parameter.
I think I chose `max_rounds = 2` for the zero-shot results listed in Table 3 in the paper.
The choice of this hyperparameter depends on the actual application scenario, e.g., if the target span is short relative to the entire vireo, or if the model is fine-tuned on this dataset, you should consider using larger `max_rounds`, as discussed in paragraph **"The Effect of Recursive Grounding Rounds"** in section 5.1.
